apps/extension/src/core/handlers/Extension.ts

```diff
@@ -86,11 +86,7 @@ export default class Extension extends ExtensionHandler {
             .filter(([, site]) => site.connectAllSubstrate)
             .forEach(async ([url, autoAddSite]) => {
               const newAddresses = Object.values(addresses)
-                .filter(
-                  ({ json: { address, meta } }) =>
-                    meta.origin !== AccountTypes.WATCHED &&
-                    !autoAddSite.addresses?.includes(address)
-                )
+                .filter(({ json: { address } }) => !autoAddSite.addresses?.includes(address))
```

Comment:
Because we might have `connectAllSubstrate` on other sites in future, we should still do the check for `meta.origin !== AccountType.Watched` here if the site is not the Portal